tools/fastdds/discovery/parser.py

```diff
@@ -34,45 +101,130 @@ def __init__(self, argv):
+                    stdout=subprocess.PIPE,
+                    universal_newlines=True
+                )
+                print(self.__edit_tool_text(result_ex.stdout, tool_name))
+                aux_argument = True
+            # Check for version argument
+            if aux_args.version:
+                result = subprocess.run(
+                    [tool_path, str(command_to_int[Command.SERVER]), '-v'],
+                    stdout=subprocess.PIPE,
+                    universal_newlines=True
+                )
+                aux_argument = True
+            if aux_argument:
+                raise SystemExit(0)
+
+            if not argv or (argv and argv[0] not in [command.value for command in Command]):
+                # Conventional server creation
+                command_int = command_to_int[Command.SERVER]
+                args_for_cpp = [str(command_int)] + sys.argv[2:]
+                result = subprocess.run([tool_path] + args_for_cpp)
                 if result.returncode != 0:
-                    sys.exit(result.returncode)
+                    raise SystemExit(result.returncode)
             else:
-                # Call the tool
-                result = subprocess.run([tool_path] + argv)
-                if result.returncode != 0:
-                    sys.exit(result.returncode)
+                if os.name == 'nt':
+                    # DS_AUTO not supported in Windows
+                    print('DS_AUTO not supported in Windows')
+                    raise SystemExit(1)
+                daemon_args, unknown_args = parse_daemon_arguments(argv)
+                try:
+                    command_int = command_to_int[Command(daemon_args.command)]
+                except:
+                    print('Unknown command')
+                    raise SystemExit(1)
+
+            # Daemon commands
+            domain = 0
+            if daemon_args.domain is not None:
+                domain = daemon_args.domain
+            else:
+                domain = self.__get_domain_id_from_env()
+            args_for_cpp = [str(command_int), '-d', str(domain)]
+            for unknown_arg in unknown_args:
+                args_for_cpp.append(unknown_arg)
+
+            if command_int == command_to_int[Command.SHUTDOWN]:
+                if not self.__is_daemon_running():
+                    print('The Fast DDS daemon is not running.')
+                    raise SystemExit(0)
+                print(client_cli.stop_all_request(get_sig_idx(signal.SIGTERM)))
+                for p in Path(self.__shm_dir()).glob("*_servers.txt"):
+                    p.unlink()
+                self.__stop_daemon()
+            elif command_int == command_to_int[Command.AUTO] or command_int == command_to_int[Command.START]:
+                self.__start_daemon(tool_path)
+                self.__add_remote_servers_to_args(args_for_cpp)
+                print(client_cli.run_request_nb(domain, args_for_cpp))
```

Comment:
Right now if the server creation fails (`Error starting Server:...` is shown in the output) it will successfully return. 
Should we regex the output from `client_cli.run_request_nb(domain, args_for_cpp)` in search of `Error starting Server` and throw an exception if we detect that in order to unsucessfully finish the process (`sys.exit(1)`)?